apps/aefate/src/aefa_fate_eval.erl

```diff
@@ -5,345 +5,457 @@
 -module(aefa_fate_eval).
 -export([eval/2]).
 
-eval('RETURN', EngineState) ->
-    aefa_fate_op:return(EngineState);
+eval('RETURN',  #{ gas := Gas } = EngineState) ->
+    EngineState1 = EngineState#{ gas => Gas - 2},
```

Comment:
Would it not be nicer and shorter to write:
`aefa_fate_op:return( EngineState#{ gas => Gas - 2}).`
Saves a few lines here and there.

or a little wrapper decrease_gas(2, EngineState).
A matter of taste, of course. Advantage for such a wrapper is that you can check in  the wrapper that gas does not get negative... but now you check in return, which is fine as well.

Reply:
Fixed